### DISCUSSION TITLE
merge multible inputs

### DISCUSSION THREAD

**Discussion by arpu** [zeux/meshoptimizer#276] [Ideas]
Hey,

would it be possible to have multiple inputs and merge  all? 
my use case is export a .vox file in MagicaVoxel to .obj Files -> multible .obj files 
 would be good to have a way to input all this in gltfpack and get all merged in one glb

**Comment by arpu** [comment]
found a good solution for this with @donmccurdy https://github.com/donmccurdy/glTF-Transform

**Comment by zeux** [comment]
Yeah I think it's probably better to use other tools for merging for now. We've discussed this on some other issue at some point, the summary is:

- There are some technical barriers in gltfpack implementation wrt making merging possible, as parts of the input tree are reused directly and merging complicates the situation with indexing
- While it would be easy to support merging just for .obj, gltfpack is a glTF-focused tool and this divergence in implementation wouldn't be desirable
- Additionally, merging often requires other scene transforms such as placing the objects at different locations, combining materials into variants, etc
- Some use cases want to merge multiple files as different scenes, not just as objects within one scene.

Depending on the requirements you could use https://github.com/donmccurdy/glTF-Transform, https://github.com/facebookincubator/glTFVariantMeld or other glTF processing tools before handing the input to gltfpack; gltfpack will try its best to give enough options to preserve semantics (e.g. it now preserves multiple scenes, preserves variants, etc.)
